### PR TITLE
Add osqp_vendor as a subtree

### PR DESCRIPTION
**Pull request by esteve**

### CODE REVIEW

**Comment by esteve** [comment]
@fred-apex-ai @mitsudome-r I'm submitting this for review separate from #14 so that perhaps the issue with `osqp_interface` not being discovered by colcon in https://github.com/fred-apex-ai/autoware.iv.universe/tree/port-osqp-interface-squashed mentioned in https://github.com/tier4/AutowareArchitectureProposal/issues/84#issuecomment-711078959 can be investigated separately and also to keep the commit history in #14 clean from subtree additions.

File: vendor/osqp_vendor/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# osqp_vendor
```
Comment (fred-apex-ai) [inline vendor/osqp_vendor/README.md:1]: I added a small reference to the source of this repo

```
added as `git` subtree from https://github.com/tier4/osqp_vendor to make `osqp` from https://github.com/oxfordcontrol/osqp available in ROS2.
```

**Review by fred-apex-ai — APPROVED** [review]
I tested the installation and verified once more that I can consume this from osqp_interface. Apart from the small doc addition, it looks great. Thanks for opening this PR to keep #14 cleaner @esteve

File: vendor/osqp_vendor/README.md (hunk 1 above)
Comment (esteve) [reply vendor/osqp_vendor/README.md]: Thanks, added in f282c80

**Comment by fred-apex-ai** [comment]
@mitsudome-r or @esteve if you also have the rights, please go ahead and merge

**Comment by esteve** [comment]
@mitsudome-r I had to enable "rebase and merge" for this repository so that the commit history of the git subtree is preserved.

**Comment by esteve** [comment]
@fred-apex-ai thanks for the review. I went ahead and merged this PR so that you can rebase #45 and we can merge it as soon as possible.

**Pull request merged**